src/db.rs

```diff
@@ -62,7 +65,18 @@ impl DbInner {
         key: &[u8],
         options: &ReadOptions,
     ) -> Result<Option<Bytes>, SlateDBError> {
-        let snapshot = self.state.read().snapshot();
+        let snapshot = if let Some(s) = options.snapshot {
+            self.manifest_store
```

Comment:
If we're given a snapshot, I don't think we need to read from the manifest. We can use the snapshot directly.